logstreams/src/main/java/io/zeebe/logstreams/impl/snapshot/fs/FsSnapshotController.java

```diff
@@ -58,26 +58,37 @@ public void takeSnapshot(SnapshotMetadata metadata) throws Exception {
   }
 
   @Override
-  public SnapshotMetadata recover(long commitPosition, int term, Predicate<SnapshotMetadata> filter)
-      throws Exception {
-    SnapshotMetadata metadata = SnapshotMetadata.createInitial(term);
+  public StateSnapshotMetadata recover(
+      long commitPosition, int term, Predicate<StateSnapshotMetadata> filter) throws Exception {
+    StateSnapshotMetadata metadata = StateSnapshotMetadata.createInitial(term);
 
     final ReadableSnapshot lastSnapshot = storage.getLastSnapshot(name);
     if (lastSnapshot != null) {
-      final SnapshotMetadata recovered =
-          new SnapshotMetadata(lastSnapshot.getPosition(), lastSnapshot.getPosition(), term, true);
+      final StateSnapshotMetadata recovered =
+          new StateSnapshotMetadata(
+              lastSnapshot.getPosition(), lastSnapshot.getPosition(), term, true);
 
-      if (filter.test(recovered)) {
-        lastSnapshot.recoverFromSnapshot(resource);
-        metadata = recovered;
-      }
+      lastSnapshot.recoverFromSnapshot(resource);
```

Comment:
We don't hold the same information (e.g. the last written event position + term). I could implement I guess, but I was trying to limit how much functionality/code to add for the Fs package since we'll throw it away. I'll happily add it if you want :)